Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -0,0 +1,109 @@
+﻿using UnityEngine;
+using Reactor.Utilities.Attributes;
+using System;
+using System.Collections;
+using Il2CppInterop.Runtime.Attributes;
+using Neuro.Utilities;
+using Neuro.Events;
+using Neuro.Cursor;
+
+namespace Neuro.Impostor;
+
+[RegisterInIl2Cpp, ShipStatusComponent]
+public sealed class ImpostorHandler : MonoBehaviour
+{
+    public ImpostorHandler(IntPtr ptr) : base(ptr) { }
+
+    public static ImpostorHandler Instance { get; private set; }
+
+    private static Vent previousVent = null;
+
+    private void Awake()
+    {
+        if (Instance)
+        {
+            NeuroUtilities.WarnDoubleSingletonInstance();
+            Destroy(this);
+            return;
+        }
+
+        Instance = this;
+    }
+
+    [HideFromIl2Cpp]
+    public IEnumerator CoStartVentOut()
+    {
+        Vent original = Vent.currentVent;
+        previousVent = original;
+        while (true)
+        {
+            yield return CoTryMoveToVent();
+            bool crewmateSpotted = false;
+
+            foreach (PlayerControl player in PlayerControl.AllPlayerControls)
+            {
+                if (player == PlayerControl.LocalPlayer) continue;
+                if (player.Data.IsDead) continue;
+                if (player.Data.Role.IsImpostor) continue;
+
+                // TODO: implement pathfinding to improve decision making
+                if (Visibility.IsVisible(player.GetTruePosition()))
+                {
+                    Info("Spotted a crewmate, trying another vent");
+                    crewmateSpotted = true;
+                    yield return CoTryMoveToVent();
+                    break;
+                }
+            }
+
+            // avoid exiting from the vent we entered
+            if (!crewmateSpotted && Vent.currentVent != original)
+            {
+                HudManager.Instance.ImpostorVentButton.DoClick();
+                InGameCursor.Instance.Hide();
```

Comment:
If the coroutine is interrupted (which can only happen if a meeting is called, won't the cursor stay on screen?)

Reply:
Although that might not be a problem since we're using the cursor in the meeting as well